modules/edu.gemini.seqexec.engine/src/main/scala/edu/gemini/seqexec/engine/Engine.scala

```diff
@@ -31,55 +34,47 @@ class Engine[D: ActionMetadataGenerator, U](implicit ev: ActionMetadataGenerator
     * It's named `Handle` after `fs2.Handle` in order to give a hint in a future
     * migration.
     */
-  type Handle[A] = HandleStateT[Task, A]
+  type Handle[A] = HandleStateT[IO, A]
   // Helper alias to facilitate lifting.
   type HandleStateT[M[_], A] = StateT[M, StateType, A]
 
-
-  // The `Catchable` instance of `Handle`` needs to be manually written.
-  // Without it it's not possible to use `Handle` as a scalaz-stream process effects.
-  implicit def engineInstance: Catchable[Handle] =
-    new Catchable[Handle] {
-      def attempt[A](a: Handle[A]): Handle[Throwable \/ A] = a.flatMap(
-        x => Catchable[Task].attempt(Applicative[Task].pure(x)).liftM[HandleStateT]
-      )
-      def fail[A](err: Throwable): Handle[A] = Catchable[Task].fail[A](err).liftM[HandleStateT]
-    }
-
   /*
-   * HandleP is a Process which has as a side effect a State machine inside a Task, which can produce other
-   * Processes as output.
+   * HandleP is a Stream which has as a side effect a State machine inside a IO, which can produce other
+   * Streams as output.
+   *
    * Its type parameters are:
    * A: Type of the output (usually Unit)
    * D: Type of the user data included in the state machine state.
    *
    * Making it final causes an error: "The outer reference in this type test cannot be checked at run time"
    */
-  case class HandleP[A](run: Handle[(A, Option[Process[Task, EventType]])])
+  case class HandleP[A](run: Handle[(A, Option[Stream[IO, EventType]])])
```

Comment:
Just a general note, when using tuples you should always put any free type variable in the right most position; i.e., `(Foo, A)` rather than `(A, Foo)` because the available instances assume right biasing.

Reply:
what are the available instances?

Reply:
Do you think we should flip this to be 
```scala
case class HandleP[A](run: Handle[(Option[Stream[IO, EventType]], A)])
```

Reply:
I wouldn't worry about it right now. It was just an observation.

Reply:
Is it also true for higher arities? `(A, B, C, ?)`

Reply:
In principle yes, but Cats doesn't define those instances. `(A, B, C, ?)` is isomorphic to `((A, B, C), ?)` so you can always coerce it into that shape temporarily if you need to.